src/mbgl/map/map.cpp

```diff
@@ -509,7 +509,7 @@ CameraOptions Map::cameraForLatLngs(const std::vector<LatLng>& latLngs, optional
         scaleX -= (padding->left + padding->right) / width;
         scaleY -= (padding->top + padding->bottom) / height;
     }
-    double minScale = ::fmin(scaleX, scaleY);
+    double minScale = (isnan(scaleX) && isnan(scaleY)) ? INFINITY : ::fmin(scaleX, scaleY);
```

Comment:
Instead of checking for NaN here, I think perhaps we should condition some of the logic above on whether `width` or `height` is zero.